scripts/apps/authoring/authoring/directives/AuthoringDirective.ts

```diff
@@ -850,12 +853,16 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
             $scope.autosave = function(item, timeout) {
                 $scope.dirty = true;
                 angular.extend($scope.item, item); // make sure all changes are available
+                return coreApplyMiddleware(onChangeMiddleware, {item: $scope.item, original: $scope.origItem}, 'item')
```

Comment:
Could you talk me through how your implementation works? I don't understand why it calls a middleware before AND after autosave. I assume it should fire on change, but you are using autosave as a debouncer?

Reply:
right autosave is called whenever user makes some changes so I use it to trigger the middleware.

Reply:
Could you also answer the first part of the question? :slightly_smiling_face:

Reply:
yep that's because first set of middlewares update the item before it gets stored. second set updates schema based on item changes. so to make sure all middlewares with item changes are done before the schema middlewares get called

Reply:
Are you updating anything in the item before it gets stored or just put the middleware just so it's there for the future?

Reply:
atm it's there for future, won't need it after all